packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -683,6 +687,16 @@ class FlutterError extends Error with DiagnosticableTreeMixin implements Asserti
         skipped.add('class ${frameLine.className}');
       } else if (filteredPackages.contains(frameLine.packageScheme + ':' + frameLine.package)) {
         skipped.add('package ${frameLine.packageScheme == 'dart' ? 'dart:' : ''}${frameLine.package}');
+      } else if (frameLine.packageScheme == 'package' && frameLine.package == 'flutter') {
+        if (visited.add('flutter/${frameLine.packagePath}:${frameLine.line}:${frameLine.column}')) {
+          if (skippedFromFlutterCount > 0) {
+            result.add('(elided $skippedFromFlutterCount frame${skippedFromFlutterCount == 1 ? '' : 's'} from package:flutter)');
```

Comment:
Maybe: 'elided 2 revisited frames ...' or something similar so that there's some explanation of what was left out.

Reply:
Talked offline.  We should figure out what the best thing for a user is here.  It might just be chopping the stack N frames of flutter code after user code.

Reply:
Would IDE users ever want to expand out the elided frames? If so, it could be worth it to expose the elided frames with enough structure that IDEs can expose expanding out the elided frames.

The ideal solution would be that an IDE received this data with enough structure to expand and collapse blocks of elided frames if requested by an IDE user who wants to see the elided frames.

I do like placing the logic of what frames should start out elided in package:flutter instead of having intellij, devtools, and VSCode create their own logic about what frames to elide when displaying flutter error messages.
It would be even better if consistent frame black boxing was also applied to stack traces displayed by IDEs but that would be a hard problem to solve.

Reply:
I personally don't think it's ever useful to see all of these frames.  Or, if it is, it's so rare that it's worth just instrumenting the code differently in the framework by hand.

Do the IDEs use the defaultStackFilter, or do they replace it?

Reply:
IDEs currently use defaultStackFilter. If eliding frames gets aggressive enough that users sometimes need to see the frames, we could revisit.

Reply:
If people need to debug something tricky there's always the option of adding breakpoints/break-on-all-exceptions where they'd get the full stack in the debugging tools.

In VS Code we do collapse multiple frames from SDKs that one (eg. "<105 frames from Flutter SDK>") but you can click-to-expand.

I suspect few people would miss these frames being removed from here (in VS Code at least) if they're still there in the debugger call stack.

Reply:
Debuggers are more or less useless when debugging this code, because stepping through what we do even for a single frame takes hours. Stack inspection and printf-debugging is really the only reasonable way I've found to debug errors deep in the framework.

Reply:
Ah, fair enough! I thought break-on-all-exceptions would break you in a good place but I've really no experience trying to debug these sorts of issues so I'll defer to the experts :-)

Jacob's original idea sounds good though - if all stack frames are provided to the IDE in a structured way with flags on which ones are reasonable to elide, then for IDEs that have an easy way to collapse them (but still make them accessible - similar to VS Code's call stack), that might be best of both worlds (for locations where interaction isn't possible, the whole list could be shown, or have a user preference - similar to how we let users device whether to have Framework libraries debuggable or not).

Reply:
Generally the stack traces I'm trying to divine from are on GitHub, no IDEs in sight. :-)